drivers/spi/spi_nrfx_spim.c

```diff
@@ -599,15 +618,19 @@ static int spi_nrfx_init(const struct device *dev)
 			    nrfx_isr, nrfx_spim_##idx##_irq_handler, 0);       \
 	}								       \
 	IF_ENABLED(SPI_BUFFER_IN_RAM,					       \
-		(static uint8_t spim_##idx##_buffer			       \
+		(static uint8_t spim_##idx##_tx_buffer			       \
+			[CONFIG_SPI_NRFX_RAM_BUFFER_SIZE]		       \
+			SPIM_MEMORY_SECTION(idx);			       \
+		 static uint8_t spim_##idx##_rx_buffer			       \
 			[CONFIG_SPI_NRFX_RAM_BUFFER_SIZE]		       \
```

Comment:
Since the `SPI_NRFX_RAM_BUFFER_SIZE` option applies now to both TX and RX, it would be good to also update its description as it still mentions transmitting from flash only.